Mage/Archivist/Archivist.cs

```diff
@@ -238,16 +238,26 @@ private void ValueDel(string topic, JObject index)
 		{
 			// Check if value already exists
 			string cacheKeyName = CreateCacheKey(topic, index);
-			VaultValue cacheValue = GetCacheValue(cacheKeyName);
-			if (cacheValue == null)
+			VaultValue cacheValue;
+
+			// NOTE: even though some of these operations lock already, we put them inside this
+			// lock to ensure there is no time inconsistencies if things happen too fast.
+			lock ((object)cache)
 			{
-				Logger.Warning("Could not delete value (doesn't exist): " + cacheKeyName);
-				return;
+				cacheValue = GetCacheValue(cacheKeyName);
+				if (cacheValue == null)
+				{
+					// If it doesn't exist, create a new vault value
+					cacheValue = new VaultValue(topic, index);
+					cache.Add(cacheKeyName, cacheValue);
+				}
+				else
```

Comment:
there is not need for an else here, set deletes before setting to destroy existing tomes etc
however the delete is the main operation here so it being in the else is not required